src/core/client/stream/tabs/Comments/PermalinkView/PermalinkViewContainer.tsx

```diff
@@ -1,110 +1,197 @@
 import { PermalinkViewContainer_viewer as ViewerData } from "coral-stream/__generated__/PermalinkViewContainer_viewer.graphql";
 
-import PermalinkView from "./PermalinkView";
+import CommentEnteredSubscription from "../Stream/AllCommentsTab/CommentEnteredSubscription";
+import ConversationThreadContainer from "./ConversationThreadContainer";
+
+import styles from "./PermalinkViewContainer.css";
 
-interface PermalinkViewContainerProps {
+interface Props {
   comment: CommentData | null;
   story: StoryData;
   settings: SettingsData;
   viewer: ViewerData | null;
   setCommentID: SetCommentIDMutation;
-  pym: PymChild | undefined;
-  eventEmitter: EventEmitter2;
 }
 
-class PermalinkViewContainer extends React.Component<
-  PermalinkViewContainerProps
-> {
-  private showAllComments = (e: MouseEvent<any>) => {
-    ViewFullDiscussionEvent.emit(this.props.eventEmitter, {
-      commentID: this.props.comment && this.props.comment.id,
-    });
-    void this.props.setCommentID({ id: null });
-    e.preventDefault();
-  };
-  private getShowAllCommentsHref() {
-    const { pym } = this.props;
-    const url = (pym && pym.parentUrl) || window.location.href;
-    return getURLWithCommentID(url, undefined);
-  }
+const PermalinkViewContainer: FunctionComponent<Props> = (props) => {
+  const { comment, story, viewer, settings, setCommentID } = props;
+  const { pym, eventEmitter } = useCoralContext();
+
+  const subscribeToCommentEntered = useSubscription(CommentEnteredSubscription);
 
-  public componentDidMount() {
-    if (this.props.pym) {
-      setTimeout(() => this.props.pym!.scrollParentToChildPos(0), 100);
+  useEffect(() => {
+    if (comment?.id) {
+      const disposable = subscribeToCommentEntered({
+        storyID: story.id,
+        ancestorID: comment.id,
+        liveDirectRepliesInsertion: true,
+        storyConnectionKey: "Stream_comments",
+      });
+      return () => {
+        disposable.dispose();
+      };
     }
-  }
+    return;
+  }, [comment?.id, story.id, subscribeToCommentEntered]);
 
-  public render() {
-    const { comment, story, viewer, settings } = this.props;
-    return (
-      <PermalinkView
-        viewer={viewer}
-        story={story}
-        comment={comment}
-        settings={settings}
-        showAllCommentsHref={this.getShowAllCommentsHref()}
-        onShowAllComments={this.showAllComments}
-      />
-    );
-  }
-}
+  useEffect(() => {
+    if (pym) {
+      setTimeout(() => pym.scrollParentToChildPos(0), 100);
+    }
+  }, [pym]);
+
+  const onShowAllComments = useCallback(
+    (e: MouseEvent<any>) => {
+      ViewFullDiscussionEvent.emit(eventEmitter, {
+        commentID: comment && comment.id,
+      });
+      void setCommentID({ id: null });
+      e.preventDefault();
+    },
+    [comment, eventEmitter, setCommentID]
+  );
+
+  const showAllCommentsHref = useMemo(() => {
+    const url = (pym && pym.parentUrl) || window.location.href;
```

Comment:
More easily written:

```suggestion
    const url = pym?.parentUrl || window.location.href;
```